integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DistributedRDDUtils.scala

```diff
@@ -47,10 +46,24 @@ object DistributedRDDUtils {
                                               .getDistributable.getSegment.getIndexSize)
     val executorCache = DistributedRDDUtils.executorToCacheSizeMapping
     // check if any executor is dead.
-    val invalidExecutors = executorCache.keySet().asScala.diff(executorsList)
-    if (invalidExecutors.nonEmpty) {
+    val invalidHosts = executorCache.keySet().asScala.diff(executorsList.keySet)
+    if (invalidHosts.nonEmpty) {
       // extract the dead executor host name
-      DistributedRDDUtils.invalidateExecutors(invalidExecutors.toSeq)
+      DistributedRDDUtils.invalidateHosts(invalidHosts.toSeq)
+    } else {
```

Comment:
I think it should not be a condition here, just check for invalid executors and invalidate them. No need to check for invalid host because there is a chance of losing one executor from host but will not be found in this logic.

Reply:
removed this code from else part